### DISCUSSION TITLE
😲這太強啦!AI幫你寫程式(最牛插件).md

### DISCUSSION THREAD

**Discussion by JunTingLin** [1101-GS4719-C/Python-Community-GS4719-C#544] [General]
# 😲這太強啦!AI幫你寫程式(最牛插件).md
![image](https://user-images.githubusercontent.com/92431095/161313076-056f51b9-20f2-48a0-a8ae-887506731ccc.png)

## GitHub Copilot是啥?
去年6 月 GitHub 與 Open AI 合作推出「GitHub Copilot」AI 編碼工具，Copilot 能根據上下文自動補好程式碼，包括文檔字符串、註釋、函數名稱、程式碼，只要編碼者給提示，AI 工具就可補好完整函數。😁

![GitHub](https://buzzorange.com/techorange/wp-content/uploads/sites/2/2021/07/GitHub-Copilot-2.gif)

## GitHub Copilot支援哪種語言
基本上你想的到的語言通通都支援~~(但現行預覽版在 Python、JavaScript、TypeScript、Ruby 及 Go 等程式語言上得表現最為出色)

GitHub 表示 Copilot 支援 Java、C、C++ 和 C# 等語言多行程式碼，並增加 Neovim 和 JetBrains IDE，包含支援開發者常用的 IntelliJ IDEA、PyCharm 等編輯器。

## 哪裡可以使用
像pycharm、vscode都可以安裝這款服務插件🛠

## 安裝流程
這邊以vscode做示範
1. 到延伸模組->輸入GitHub Copilot
![螢幕擷取畫面 2022-03-28 192819](https://user-images.githubusercontent.com/92431095/161312632-91acb09a-8069-4a68-8c90-ee4ca83e07d8.png)

2. 按下join the waitlist
![GitHub](https://miro.medium.com/max/1400/1*yEgiS8UMyVKPY4czYIZ3KA.png)
就會導向[copilot官網](https://copilot.github.com/)用github帳號授權

3. 過幾天後就會收到回信了
![image](https://user-images.githubusercontent.com/92431095/161312767-219402fd-ee44-4135-8f8f-e4d037944f7a.png)

4. 假如還是沒有辦法使用，記得檢查一下看是不是vscode沒有登錄github帳號，登入後才能連到GitHub Copilot的server
![螢幕擷取畫面 2022-04-02 003410](https://user-images.githubusercontent.com/92431095/161312648-3c0b0b40-0706-4357-954f-8944a4e8f2bf.png)

## 影片-用AI寫AI
雖然沒接觸過機器學習，但大概知道很程式碼是很制式的，比如說取樣完下一步應該要幹嘛，此時Copilot就超級好用，一直按tab鍵接受建議，重點是準度還很高，少了程式設計師很多時間⏰
[![用 AI 來做 AI：GitHub Copilot 初體驗](https://res.cloudinary.com/marcomontalbano/image/upload/v1648833846/video_to_markdown/images/youtube--RPOlHiMHZTE-c05b58ac6eb4c4700831b2b3070cd403.jpg)](https://www.youtube.com/watch?v=RPOlHiMHZTE&ab_channel=%E5%A4%A7%E9%AD%94%E8%A1%93%E7%86%8A%E8%B2%93%E5%B7%A5%E7%A8%8B%E5%B8%AB "用 AI 來做 AI：GitHub Copilot 初體驗")
## 結語
試用了幾天，不能說超讚，只能說超級讚，除了上面提到的寫python AI、機器學習很有幫助，他還會根據我的註解給我建議，預判我下一步要寫的內容，而且超準，用口述很難敘述他的好，大家有空一定要去裝裝看✨
## 參考資料
+ [
程式設計師用 AI 寫 AI 程式碼，「搞死」其他程式設計師？](https://technews.tw/2021/11/10/github-copilot-ai-coder/)

+ [【工程師生產力神器】OpenAI、Github 推新 AI 工具 Copilot，輸入「文字」就可轉譯成「程式碼」](https://buzzorange.com/techorange/2021/07/08/microsoft-openai-github-copilot/)

**Comment by 11timyyy** [comment]
趕緊來下載看看!

**Comment by JSHT** [comment]
這個團隊就是之前開發 GPT-3 的團隊，後來也開發了 API 供企業或開發者使用，可以廣泛用來文章生成、程式碼生成、公式生成等。
[GPT-3 Design Hype](https://prototypr.io/post/gpt-3-design-hype/) 介紹了幾個由開發者利用 GPT-3 設計的功能，
另外這邊也有使用 GPT-2 模型製作的文章生成器，可以透過自己輸入的文字來生成後續的句子，可以玩玩看。
[Write With Transformer](https://transformer.huggingface.co/doc/gpt2-large)

**Reply by JSHT** [reply]
對了，copilot 就是使用 GPT-3 做開發的喔。

**Reply by JSHT** [reply]
此外，[Tabnine](https://marketplace.visualstudio.com/items?itemName=TabNine.tabnine-vscode) 這個插件也是運用 AI 預測開發者要輸入的內容，使用的是 GPT-2。

**Comment by leon10456** [comment]
喔喔喔 馬上來用用看

**Comment by ThomasMeng0515** [comment]
這樣確實能節省許多花在機器學習上制式化程式碼的時間，感謝分享

**Comment by hapiwangy** [comment]
自然語言處理的領域在目前就是非常熱門的一環，這個技術的發表讓我們見識到了人工智慧仍然有很大的發展性。

**Comment by yuhao112** [comment]
這技術太扯了...雖然要學習應用他也是一大難點就是了

**Comment by liangyinglly** [comment]
這個技術也太省時了吧 但好像不太適合菜鳥使用 像我這種程設菜鳥可能連下一步是什麼都不是很確定 哈

**Comment by Lai27330** [comment]
這個新技術好方便 感覺可以節省許多因為拼字錯誤而需debug的時間 編寫複雜程式碼的時候也不會因為缺打了小細節而花費大量時間檢查 感覺可以下載來試試看

**Comment by ghost** [comment]
也太扯ww用AI寫AI的概念ww，當然如何寫還是人們自己決定

**Comment by sean1325418** [comment]
打程式原本就是很有邏輯的，能有這種東西好像也不奇怪，不過還是覺得開發這的人很厲害

**Comment by Ku-Hao-Chen** [comment]
謝謝分享，開發的人真的很厲害。

**Comment by PoooYen** [comment]
超適合漏東漏西的人的...感謝教學!!!!

**Comment by yi-1111** [comment]
太神了，這樣就可以減少熬夜debug了

**Comment by d26488399** [comment]
感謝分享，感覺可以提升寫程式效率

**Comment by HsuChen88** [comment]
馬上來試試看

**Comment by handsomeniko** [comment]
好猛的插件

**Comment by sys511613** [comment]
用這個插件一段時間了
只能說真的很厲害
也節省了不少時間
但是有時候這個東西生出來的code還是會有一些小bug
或是寫出來根本不是自己原本想要的東西
需要自己再多檢查一下~

**Comment by Sazmafuji** [comment]
wow，這已經是AI可以寫出更強大AI第一步ㄟ

**Comment by henrywen547** [comment]
要駕馭它一開始還是要有一點邏輯，知道要它做甚麼事，但怕用了這個就回不去當初自己編寫程式的感覺了

**Comment by minniewei** [comment]
但怕就是太過便利導致我們程式語法全部忘光光

**Comment by Zoe06060606** [comment]
這感覺可以減少超多debug的時間
每次寫程式最討厭就是寫好了一大段結果run的時候一堆紅線🥲De到哭出來

**Comment by Valentinaa0214** [comment]
好酷又實用的插件，謝謝發文者這麼用心又詳細的介紹。
但不知道這麼便利的技術會不會帶來甚麼倫理問題。
或許說不定哪天AI能夠讓自己創造自己...

**Comment by Jzd314** [comment]
這個工具對於初期學習程式語言的人或許會很有幫助，遇到出現問題的地方可以參考看看機器是怎麼寫的
